docs/markdown/DevGuide_Data_Memory.md

```diff
@@ -0,0 +1,85 @@
+# Data Structures
+In Inviwo the main core data structures (`Volume`, `Image`, `Layer`, `Mesh`, `Buffer`) use a pattern of handles and representations. The volume data structure for example has a handle class called `Volume`. The handle class by itself does not have any data, it only stores metadata, and a list of representations. The representations is where the actual data is stored. The following first explains the basic data structures and how they relate, then shows what different representations exist and how to use them.
```

Comment:
one could note that the basic Data structures here are the Volume, Layer, and Buffer. Image and Mesh are then containers of Layers, and Buffers, respectively. 

And on a side note, that is a bit strange. Both the Image and the Layer as Representation, and the image representation refers to the layer representations. As an example:
```
//      ┌───────────────┐
//      │     Image     ├────┬────────────────────────┬──────────────────────────┐
//      └───────┬───────┘    │     ┌───────────────┐  │   ┌───────────────┐      │  ┌───────────────┐
//              │            └─────▶  Color Layer  │  └───▶  Depth Layer  │      └──▶  Pick Layer   │
//              │                  └───────┬───────┘      └─────────┬─────┘         └───────┬───────┘
//              │                          │                        │                       │
//              │                          │                        │                       │
//         ┌────┴────┐                 ┌───┴─────────┐          ┌───┴────────┐          ┌───┴─────────┐
//         │         │                 │             │          │            │          │             │
//         │         │                 │     ┌───────▼───────┐  │    ┌───────▼───────┐  │     ┌───────▼───────┐
//         │         │                 │  ┌──▶   Layer GL    │  │ ┌──▶   Layer GL    │  │  ┌──▶   Layer GL    │
//         │ ┌───────▼───────┐         │  │  └───────────────┘  │ │  └───────────────┘  │  │  └───────────────┘
//         │ │   Image GL    ├─────────┼──┴─────────────────────┼─┴─────────────────────┼──┘
//         │ └───────────────┘         │                        │                       │
//         │                           │                        │                       │
//         │                   ┌───────▼───────┐        ┌───────▼───────┐       ┌───────▼───────┐
//         │                ┌──▶   Layer RAM   │     ┌──▶   Layer RAM   │    ┌──▶   Layer RAM   │
// ┌───────▼───────┐        │  └───────────────┘     │  └───────────────┘    │  └───────────────┘
// │   Image RAM   ├────────┴────────────────────────┴───────────────────────┘
// └───────────────┘
```